src/lib/util/corefile.cpp

```diff
@@ -1292,17 +1296,17 @@ std::string core_filename_extract_base(const char *name, bool strip_extension)
     filename end with the specified extension?
 -------------------------------------------------*/
 
-int core_filename_ends_with(const char *filename, const char *extension)
+bool core_filename_ends_with(const std::string &filename, const std::string &extension)
 {
-	int namelen = strlen(filename);
-	int extlen = strlen(extension);
-	int matches = TRUE;
+	auto namelen = filename.length();
+	auto extlen = extension.length();
+	bool matches = true;
 
 	/* work backwards checking for a match */
 	while (extlen > 0)
 		if (tolower((UINT8)filename[--namelen]) != tolower((UINT8)extension[--extlen]))
 		{
-			matches = FALSE;
+			matches = false;
 			break;
```

Comment:
This could also be done with C++ algorithms, e.g. something like:

    auto found(std::find_if(extension.rbegin(), extension.rend(), [it = filename.rbegin()](char c) mutable { return (it == filename.rend()) || (std::tolower(c) != std::tolower(*it--)); });
    return found == extension.rbegin();